knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/impl/TemplateSnippetProcessor.java

```diff
@@ -51,10 +53,10 @@ public TemplateSnippetProcessor(Vertx vertx, TemplateEngineConfiguration configu
 
     public Observable<String> processSnippet(final HtmlFragment fragment, TemplateEngineRequest request) {
         return fragment.getServices()
-                .filter(serviceEntry -> serviceEntry.canServeMethodType(request.getServerRequestMethod()))
+                .filter(serviceEntry -> serviceEntry.canServeRequest(fragment, request))
```

Comment:
Do we have some test or scenario what will happen if serviceEntry#canServeRequest() will return false ?